contracts/examples/NFTExample.sol

```diff
@@ -38,29 +35,17 @@ contract NFTExample is
             super.supportsInterface(interfaceId);
     }
 
-    function setTokenURI(
-        uint256 tokenId,
-        ctUint64 _itTokenURI,
-        bytes calldata _itSignature
-    ) public {
-        _requireOwned(tokenId);
-        address owner = _ownerOf(tokenId);
-        if (msg.sender != owner) {
-            revert ERC721IncorrectOwner(msg.sender, tokenId, owner);
-        }
-
-        itUint64 memory it;
-        it.ciphertext = _itTokenURI;
-        it.signature = _itSignature;
-
-        _setTokenURI(owner, tokenId, MpcCore.validateCiphertext(it));
-    }
-
-    function mint(address to) public onlyOwner {
+    function mint(
+        address to,
+        ctUint64[] calldata _itTokenURI,
+        bytes[] calldata _itSignature
+    ) public onlyOwner {
         uint256 tokenId = _totalSupply;
-        _mint(to, tokenId);
+        _mint(to, tokenId); // at this stage, token URI is null and can leak some information
```

Comment:
notice the comment there - it can not be leaked now it is a mandatory value for the mint operation

Reply:
True